lib/matchers/signature.ts

```diff
@@ -0,0 +1,70 @@
+import { expect } from "@jest/globals";
+import { Bool, Field, PublicKey, Signature, State } from "snarkyjs";
+import { BoolLike, FieldLike } from "../types";
+
+/**
+ * Extracts a value from a state object or returns the value itself if it's not a state object.
+ * @param val - The value to extract from.
+ * @returns The extracted value.
+ */
+function extractFromState<T extends Signature>(
```

Comment:
@gonzalobustosglob I think we should move this fn to a utils file or something, otherwise we are repeating it

Reply:
moved.